src/lib.rs

```diff
@@ -440,6 +427,52 @@ impl EventHandler for WatchEventHandler {
     }
 }
 
+struct ShareableChild(Arc<Mutex<Option<Child>>>);
+
+impl ShareableChild {
+    fn new() -> Self {
+        Self(Arc::new(Mutex::new(None)))
+    }
+
+    fn replace(&self, child: Option<Child>) {
+        let mut guard = self.0.lock().expect("can lock");
+        *guard = child;
+    }
+
+    fn wait(&self) -> bool {
+        let success = loop {
+            match self
+                .0
+                .lock()
+                .expect("can lock")
+                .as_mut()
+                .expect("is some")
+                .try_wait()
+            {
+                Ok(Some(status)) if status.success() => {
+                    break true;
+                }
+                Ok(Some(_status)) => {
+                    break false;
+                }
+                Ok(None) => std::thread::sleep(std::time::Duration::from_millis(10)),
+                Err(err) => println!("cannot wait child process: {err}"),
+            }
+        };
+
+        self.replace(None);
+
+        success
+    }
+
+    fn kill(&self) {
+        if let Some(child) = self.0.lock().expect("can lock").as_mut() {
+            let _ = child.wait();
+            let _ = child.kill();
```

Comment:
haha :grin: you actually need to do the opposite. Otherwise you will wait for the child to complete